File: sdks/typescript/packages/sdk/src/resources/webhooks.ts
import { HttpClient } from '../client';

export interface Webhook {
  id: string;
  url: string;
  events: string[];
  active: boolean;
  fail_count: number;
  created_at: string;
}

export interface CreateWebhookRequest {
  url: string;
  events: string[];
}

export class WebhookResource {
  constructor(private client: HttpClient) {}

  /**
   * Creates a new webhook endpoint.
   *
   * @example
   * ```typescript
   * const webhook = await argos.webhooks.create({
   *   url: 'https://yourapp.com/webhooks/argos',
   *   events: ['transaction.confirmed', 'wallet.created'],
   * });
   * ```
   */
  async create(req: CreateWebhookRequest): Promise<Webhook> {
    return this.client.post<Webhook>('/v1/webhooks', req);
  }

  /** Lists all webhooks. */
  async list(): Promise<Webhook[]> {
    return this.client.get<Webhook[]>('/v1/webhooks');
  }

  /** Deletes a webhook. */
  async delete(webhookId: string): Promise<void> {
    await this.client.delete(`/v1/webhooks/${webhookId}`);
  }
}

File: sdks/dart/lib/src/resources/webhooks.dart
import '../resource_base.dart';
import '../models/webhook.dart';

/// Webhook management resource.
class WebhookResource extends ArgosResource {
  WebhookResource(super.client);

  /// Creates a new webhook endpoint.
  Future<Webhook> create({
    required String url,
    required List<String> events,
  }) async {
    final json = await client.post('/v1/webhooks', {
      'url': url,
      'events': events,
    });
    return Webhook.fromJson(json);
  }

  /// Lists all webhooks.
  Future<List<Webhook>> list() async {
    final json = await client.get('/v1/webhooks');
    final list = json as List? ?? [];
    return list.map((w) => Webhook.fromJson(w as Map<String, dynamic>)).toList();
  }

  /// Deletes a webhook.
  Future<void> delete(String webhookId) async {
    await client.delete('/v1/webhooks/$webhookId');
  }
}
